### PR TITLE
Мещанов Task11

### PR DESCRIPTION
**Pull request by HugoLoud**
# Задача 11. Сравнение методов регрессии

** **Дедлайн**: 19.12.2024, 23:59
* Полный балл: 12

## Задача

- [x] Найти данные, на которых интересно будет решать задачу регрессии.
- [x] Считать данные, выполнить первичный анализ данных, при необходимости произвести чистку данных (**Data Cleaning**).
- [x] Выполнить разведочный анализ (**EDA**), использовать визуализацию, сделать выводы, которые могут быть полезны при дальнейшем решении задачи регрессии.
- [x] При необходимости выполнить полезные преобразования данных (например, трансформировать категориальные признаки в количественные), убрать ненужные признаки, создать новые (**Feature Engineering**).
- [x] Используя **подбор гиперпараметров**, **кросс-валидацию** и при необходимости **масштабирование данных**, добиться наилучшего качества предсказания для библиотечных реализаций (например, из **sklearn**) четырёх регрессионных методов.
- [x] Сравнить все обученные модели, построить графики зависимости выбранной **loss function** от времени обучения. Построить сравнительные графики точности для обученных моделей. Сделать выводы о полученных моделях в рамках решения задачи регрессии на выбранных данных.
- [x] (**+5 баллов**) Самостоятельно реализовать один из регрессионных методов, с возможностью настройки гиперпараметров. Повторить все пункты задания для собственной реализации.
- [x] (**+3 балла**) Используя два разных фреймворка автоматического машинного обучения, получить две модели для решения задачи регрессии. Сравнить полученные модели с использованными Вами.

Ссылка на colab: https://colab.research.google.com/drive/1eUTMf0UrfOZFBhNG-emhffXt0UZWH1v-#scrollTo=TGelZMCofU5-

### CODE REVIEW

**Review by Krekep — CHANGES_REQUESTED** [review]
Очень плохо. Прям очень. Вам необходимо полностью переделать задание, начиная с выбора датасета, подходящего под задачу регрессии

**Review by Krekep — APPROVED** [review]
Зачтено
